worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -781,24 +794,27 @@ private[celeborn] class Worker(
           e)
     }
     shutdown.set(true)
+    workerStatusManager.transitionState(State.Exit)
   }
 
   ShutdownHookManager.get().addShutdownHook(
     new Thread(new Runnable {
       override def run(): Unit = {
         logInfo("Shutdown hook called.")
-        exitKind match {
-          case CelebornExitKind.WORKER_GRACEFUL_SHUTDOWN =>
-            logInfo("Worker start to shutdown gracefully")
+        workerStatusManager.exitEventType match {
+          case WorkerEventType.Graceful =>
             shutdownGracefully()
-          case CelebornExitKind.WORKER_DECOMMISSION =>
-            logInfo("Worker start to decommission")
+          case WorkerEventType.Decommission =>
             decommissionWorker()
           case _ =>
-            logInfo("Worker start to exit immediately")
             exitImmediately()
         }
-        stop(exitKind)
+
+        if (workerStatusManager.exitEventType == WorkerEventType.Graceful) {
+          stop(CelebornExitKind.WORKER_GRACEFUL_SHUTDOWN)
+        } else {
+          stop(CelebornExitKind.EXIT_IMMEDIATELY)
+        }
```

Comment:
How about move this part to worker status manager

Reply:
To make this pr more focus, IMO we can do this in next pr.